### PR TITLE
Add build support files

### PR DESCRIPTION
**Pull request by ankurcha**
* Add `.gitignore` file to make sure unintended files are not added.
* Add `.travis.yaml` to do automated CI builds.
* Add `Dockerfile` to create docker image (very useful when deploying to kuberentes)

### CODE REVIEW

**Comment by ankurcha** [comment]
I signed it!

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,8 @@
+FROM maven:3-jdk-8-onbuild
+
+ENV JVM_GC_OPTS="-XX:+UseG1GC -XX:MaxGCPauseMillis=200 -XX:+UseStringDeduplication -XX:+ParallelRefProcEnabled"
```
Comment (ankurcha) [inline Dockerfile]: The idea behind putting an entry point is to make sure that we are able to put some additional Java opts.

Does that make sense? I used the pattern from the appengine-openjdk runtime docker image.

Comment (kevinmdavis) [inline Dockerfile]: I think the user could set the JAVA_OPTS environment variable and the Dockerfile could pass that directly to the java command.
```
CMD java $JAVA_OPTS -jar /usr/src/app/server/target/server-*.jar
```
and then the user can run the container by:
```
docker run -e JAVA_OPTS="-Xmx1024M -XX:+UseG1GC" "stackdriver-zipkin-server"
```

I think in this case, we don't need a separate entrypoint, and the user can pass in their JVM args. Would this approach work for your use case?

**Comment by ankurcha** [comment]
I updated the docker file as per the code review feedback.

**Comment by kevinmdavis** [comment]
Thanks! Can we add the Kubernetes docs as a separate pull request? I can go ahead and merge your other changes, and I'll work on getting the container published. After that's done, we can update the Kubernetes docs with the official container location.

**Comment by ankurcha** [comment]
I'll remove kube from here and make a new pr. Thanks!

**Comment by ankurcha** [comment]
Done! Should be good to go.

**Pull request merged**